### PR TITLE
[feat] 일기 수정 기능 구현

### PR DESCRIPTION
**Pull request by ruh0n**
## :man_shrugging: Description

- [x] 일기 수정 기능 구현

### CODE REVIEW

**Comment by ruh0n** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/boostcampwm-2024/and01-dreamDiary/157?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#158** <a href="https://app.graphite.dev/github/pr/boostcampwm-2024/and01-dreamDiary/158?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#157** <a href="https://app.graphite.dev/github/pr/boostcampwm-2024/and01-dreamDiary/157?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.dev/github/pr/boostcampwm-2024/and01-dreamDiary/157?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* **#156** <a href="https://app.graphite.dev/github/pr/boostcampwm-2024/and01-dreamDiary/156?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#155** <a href="https://app.graphite.dev/github/pr/boostcampwm-2024/and01-dreamDiary/155?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#154** <a href="https://app.graphite.dev/github/pr/boostcampwm-2024/and01-dreamDiary/154?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `dev`

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

File: core/data/src/main/java/com/boostcamp/dreamteam/dreamdiary/core/data/database/dao/DreamDiaryDao.kt [hunk 1]
```diff
@@ -44,6 +44,47 @@ interface DreamDiaryDao {
         setLabelsToDreamDiary(dreamDiaryId, labels)
     }
 
+    @Query(
+        """
+        update diary
+        set title = :title, body = :body,sleepStartAt = :sleepStartAt,sleepEndAt = :sleepEndAt, updatedAt = :updatedAt
```
Comment (ruh0n) [inline core/data/src/main/java/com/boostcamp/dreamteam/dreamdiary/core/data/database/dao/DreamDiaryDao.kt:50]: 수정해주세요!


```suggestion
        set title = :title, body = :body, sleepStartAt = :sleepStartAt, sleepEndAt = :sleepEndAt, updatedAt = :updatedAt
```

File: core/data/src/main/java/com/boostcamp/dreamteam/dreamdiary/core/data/repository/DefaultDreamDiaryRepository.kt [hunk 2]
```diff
@@ -77,6 +53,23 @@ internal class DefaultDreamDiaryRepository @Inject constructor(
...
+        labels: List<String>,
+        sleepStartAt: Instant?,
+        sleepEndAt: Instant?,
+    ) {
+        dreamDiaryDao.updateDreamDiary(
+            diaryId = diaryId,
+            title = title,
+            body = makeBody(diaryContents = diaryContents),
+            sleepStartAt = sleepStartAt ?: Instant.now(),
+            sleepEndAt = sleepEndAt ?: Instant.now(),
```
Comment (ruh0n) [inline core/data/src/main/java/com/boostcamp/dreamteam/dreamdiary/core/data/repository/DefaultDreamDiaryRepository.kt:69]: nullable하게 사용하면 안될 것 같습니다.


```suggestion
        sleepStartAt: Instant,
        sleepEndAt: Instant,
    ) {
        dreamDiaryDao.updateDreamDiary(
            diaryId = diaryId,
            title = title,
            body = makeBody(diaryContents = diaryContents),
            sleepStartAt = sleepStartAt,
            sleepEndAt = sleepEndAt,
```

File: core/data/src/main/java/com/boostcamp/dreamteam/dreamdiary/core/data/repository/DreamDiaryRepository.kt [hunk 3]
```diff
@@ -24,6 +24,15 @@ interface DreamDiaryRepository {
         sleepEndAt: Instant,
     )
 
+    suspend fun updateDreamDiary(
+        diaryId: String,
+        title: String,
+        diaryContents: List<DiaryContent>,
+        labels: List<String>,
+        sleepStartAt: Instant?,
+        sleepEndAt: Instant?,
```
Comment (ruh0n) [inline core/data/src/main/java/com/boostcamp/dreamteam/dreamdiary/core/data/repository/DreamDiaryRepository.kt:33]: 여기도 수정 해주세요




```suggestion
        sleepStartAt: Instant,
        sleepEndAt: Instant,
```

**Review by minwooch — APPROVED** [review]

**Review by easyhak — APPROVED** [review]

**Pull request merged**